src/Microsoft.AspNetCore.OData/Results/BadRequestODataResult.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Results
+{
+    /// <summary>
+    /// Represents a result that when executed will produce a Bad Request (400) response.
+    /// </summary>
+    /// <remarks>This result creates an <see cref="ODataError"/> with status code: 400.</remarks>
+    public class BadRequestODataResult : BadRequestResult
```

Comment:
>BadRequestODataResult [](start = 17, length = 21)

BadRequestODataResult, NotFoundODataResult, UnauthorizedODataResult could be have a base class to handle the "message" or "ODataError"

Reply:
#Resolved

Reply:
Not sure how that would work since these already derive from the error-specific base classes (i.e., NotFoundODataResult : NotFoundResult).

---
In reply to: [534676837](https://github.com/OData/WebApi/pull/2341#discussion_r534676837) [](ancestors = 534676837)